`interpret` support group effects

GStechschulte opened the pull request:
This PR resolves the `KeyError` in PR #716. This error was the result of no default values being computed for model terms specified as group level effects. This is because group level effects do not have a components attribute, and thus never satisfied the logic in the `set_default_values()` function of `../interpret/utils.py`. 

Now, to compute default values, all **unique** common and group specific effects variable names are identified by accessing the `components` and `factor` attributes and are put into a list. From there, the data type is determined by identifying the data types used to fit the model which is then used to determine the function call (`mean` or `mode` for numeric and categorical dtypes, respectively).

Additionally, with group specific effects, when the user (or default values are computed) passes values, it is possible these values are new unseen data. Thus, I leverage PR #693 and add the arg `sample_new_groups` to the functions `predictions`, `comparisons`, and `slopes` (and the associated plotting functions) to allow plotting of new groups.

For a working example, view my last comment in PR #716.

To do:

- [x] Add tests for new a model with group specific effects

tomicapretto commented:
I would like to ask two things: 

* Don't include anything related to the PR #716. As far as it is, this PR would add the notebook. I think it's because the branch was created from the branch in that PR. It should be created from the main branch.
* Could you add a test for the new supported cases? One reason why we didn't catch it before is that it was untested :sweat_smile:

GStechschulte commented:
Yes, you are right. I checked out this branch from Nathaniel's branch. I have removed everything from that branch now. As for tests go, I have added a model with common and group effects. I test each plotting function in `interpret` for new unseen data and non-new observed data.

tomicapretto reviewed (approved):
Looks great! You can hit the green button! (squash and merge)
